src/test/moves/rage_fist.test.ts

```diff
@@ -0,0 +1,141 @@
+      .moveset([ Moves.RAGE_FIST, Moves.SPLASH, Moves.SUBSTITUTE ])
+      .startingLevel(100)
+      .enemyLevel(1)
+      .enemyAbility(Abilities.BALL_FETCH)
+      .enemyMoveset(Moves.DOUBLE_KICK);
+
+    vi.spyOn(move, "calculateBattlePower");
+  });
+
+  it("should have 100 more power if hit twice before calling Rage Fist", async () => {
+    game.override
+      .enemySpecies(Species.MAGIKARP);
+
+    await game.classicMode.startBattle([ Species.MAGIKARP ]);
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.phaseInterceptor.to("TurnEndPhase");
+
+    expect(move.calculateBattlePower).toHaveLastReturnedWith(150);
+  });
+
+  it("should maintain its power during next battle if it is within the same arena encounter", async () => {
+    game.override
+      .enemySpecies(Species.MAGIKARP)
+      .startingWave(1);
+
+    await game.classicMode.startBattle([ Species.MAGIKARP ]);
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.toNextWave();
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.phaseInterceptor.to("BerryPhase", false);
+
+    expect(move.calculateBattlePower).toHaveLastReturnedWith(250);
+  });
+
+  it("should reset the hitRecCounter if we enter new trainer battle", async () => {
+    game.override
+      .enemySpecies(Species.MAGIKARP)
+      .startingWave(4);
+
+    await game.classicMode.startBattle([ Species.MAGIKARP ]);
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.toNextWave();
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.phaseInterceptor.to("BerryPhase", false);
+
+    expect(move.calculateBattlePower).toHaveLastReturnedWith(150);
+  });
+
+  it("should not increase the hitCounter if Substitute is hit", async () => {
+    game.override
+      .enemySpecies(Species.MAGIKARP)
+      .startingWave(4);
+
+    await game.classicMode.startBattle([ Species.MAGIKARP ]);
+
+    game.move.select(Moves.SUBSTITUTE);
+    await game.setTurnOrder([ BattlerIndex.PLAYER, BattlerIndex.ENEMY ]);
+    await game.phaseInterceptor.to("MoveEffectPhase");
+
+    expect(game.scene.getPlayerPokemon()?.customPokemonData.hitsRecCount).toBe(0);
+  });
+
+  //For some unknown reason the second Rage fist is not called. This might be due to entering a new biome
+  it.todo("should reset the hitRecCounter if we enter new biome", async () => {
+    game.override
+      .enemySpecies(Species.MAGIKARP)
+      .startingWave(10);
+
+    await game.classicMode.startBattle([ Species.MAGIKARP ]);
+
+    game.move.select(Moves.RAGE_FIST);
+    await game.setTurnOrder([ BattlerIndex.ENEMY, BattlerIndex.PLAYER ]);
+    await game.toNextWave();
```

Comment:
```suggestion
    await game.toNextTurn();
```
Because there's no `ModifierSelectPhase` when you go from wave `10 -> 11`, this causes queued up menu commands to break things since they're being executed at the wrong time (`.toNextWave()` adds menu commands to pass through the `ModifierSelectPhase` that occurs at the end of most waves (`1 -> 2`, `2 -> 3`, ...) and then runs `.toNextTurn()` after that). Since there's no MSP, you can just run `.toNextTurn()` directly.